idl/internal/thrift.y

```diff
@@ -268,38 +270,65 @@ enum_item
     ;
 
 fields
-    : /* nothing */ { $$ = nil }
+    : /* nothing */             { $$ = nil }
     | fields field optional_sep { $$ = append($1, $2) }
     ;
 
-
 field
-    : lineno docstring INTCONSTANT ':' field_required type IDENTIFIER type_annotations
+    : lineno docstring field_identifier field_required type IDENTIFIER type_annotations
         {
             $$ = &ast.Field{
-                ID: int($3),
-                Name: $7,
-                Type: $6,
-                Requiredness: $5,
-                Annotations: $8,
+                ID: $3,
+                Name: $6,
+                Type: $5,
+                Requiredness: $4,
+                Annotations: $7,
                 Line: $1,
                 Doc: ParseDocstring($2),
             }
         }
-    | lineno docstring INTCONSTANT ':' field_required type IDENTIFIER '=' const_value
-      type_annotations
+    | lineno docstring field_required type IDENTIFIER type_annotations
         {
             $$ = &ast.Field{
-                ID: int($3),
-                Name: $7,
-                Type: $6,
-                Requiredness: $5,
-                Default: $9,
-                Annotations: $10,
+                IDUnset: true,
+                Name: $5,
+                Type: $4,
+                Requiredness: $3,
+                Annotations: $6,
                 Line: $1,
                 Doc: ParseDocstring($2),
             }
         }
+    | lineno docstring field_identifier field_required type IDENTIFIER '=' const_value type_annotations
```

Comment:
I think we could reduce this from four cases to two by introducing a `default_value` pattern that just returns `nil` when `= const_value` isn't there. I didn't want to make that change in the middle of the rest of these changes, though.

Reply:
Yeah, that's a reasonable change and it's okay to leave it out of this.
I'm happy to do that in a follow up to this once this lands if you don't have the time to.